semver/parse.ts

```diff
@@ -4,6 +4,11 @@ import { isValidNumber } from "./_shared.ts";
 import { isSemVer } from "./is_semver.ts";
 import { FULL, MAX_LENGTH, NUMERICIDENTIFIER, re, src } from "./_shared.ts";
 
+/**
+ * @deprecated (will be removed in 0.211.0) parse only parses strings
+ */
+export function parse(version: SemVer): SemVer;
+export function parse(version: string): SemVer;
 /**
  * Attempt to parse a string as a semantic version, returning either a `SemVer`
  * object or throws a TypeError.
```

Comment:
I think this jsdoc block now needs to be above `function parse(version: string): SemVer;`

Reply:
done